### PR TITLE
Entrant email lists

### PR DESCRIPTION
**Pull request by jmacdotorg**
Adds a new action at /admin/email that lists email address and forum handles for authors of the current competition who have at least one non-disqualified entry.

### CODE REVIEW

**Comment by jmacdotorg** [comment]
I don't know why the tidy check fails... I have the latest Code::TidyAll, and xt/tidyall.t works for me.

**Comment by adherzog** [comment]
@jmacdotorg Try running tidyall with the --refresh-cache option and see if that makes a difference.

**Comment by jmacdotorg** [comment]
Thanks, I didn’t know about that. Running it on my repo didn’t change any files, though.

File: IFComp/lib/IFComp/Schema/Result/Comp.pm [hunk 1]
```diff
@@ -206,6 +206,7 @@ __PACKAGE__->has_many(
 
 use DateTime::Moonpig;
 use Moose::Util::TypeConstraints;
+use List::Util qw(none);
```
Comment (adherzog) [inline IFComp/lib/IFComp/Schema/Result/Comp.pm]: It doesn't look like this is used anywhere. Maybe it's left over from an earlier approach?

File: IFComp/lib/IFComp/Schema/Result/Comp.pm [hunk 2]
```diff
@@ -302,5 +303,43 @@ sub get_vote_counts_from_non_unique_ips {
     return $data;
 }
 
+sub emails {
+    my $self            = shift;
+    my $is_disqualified = 0;
```
Comment (adherzog) [inline IFComp/lib/IFComp/Schema/Result/Comp.pm]: Why is this here as a separate var, instead of just putting 0 into the query? Was it going to be a method parameter at some point?

File: IFComp/root/src/admin/email/index.tt [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+<p>This is a simple list of forum handles addresses for all authors with at least one non-disqualified entry registered with the [% current_comp.year %] IFComp.</p>
+
+<p>Note that it will likely be shorter than the email-address list, because not all authors will set a forum handle for themselves.</p>
+
+[% IF forum_handles.size > 0 %]
+<div class="well">
+<p>
+[% FOREACH forum_handle IN forum_handles %]
+[% forum_handle %]<br />
```
Comment (adherzog) [inline IFComp/root/src/admin/email/index.tt]: I'm not sure what restrictions there are for the forum handles - maybe this should be put through the html filter just in case?

**Comment by adherzog** [comment]
@jmacdotorg Please show the command invocation and output.

**Comment by adherzog** [comment]
Some POD for the new methods/actions wouldn't hurt, either.

**Comment by jmacdotorg** [comment]
@adherzog It's been a long time! Could you please check this work? I just attended to your suggested code changes.

To refresh some outstanding questions you had:

> Please show the command invocation and output.

This was in re: to my complaining about tidyall, which is no longer relevant, so please disregard.

> Some POD for the new methods/actions wouldn't hurt, either.

I agree -- but since the whole dang codebase is POD-weak, and time is short (which is my bad), I'd like to leave this embarrassingly uncovered for now, and make a separate project for PODding up the whole thing.

**Comment by adherzog** [comment]
Please make the same change regarding `$is_disqualified` in `forum_handles()` as you made in `email()`. Otherwise, it looks good to me.
